modules/configuration-annotation-processor/src/main/java/org/apache/ignite/internal/configuration/processor/ConfigurationProcessor.java

```diff
@@ -849,12 +848,12 @@ private void checkIncompatibleClassAnnotations(
         assert clazz.getAnnotation(clazzAnnotation) != null : clazz.getQualifiedName();
         assert !nullOrEmpty(incompatibleAnnotations);
 
-        Annotation incompatible = findFirst(clazz, incompatibleAnnotations);
+        Optional<? extends Annotation> incompatible = findFirstPresentAnnotation(clazz, incompatibleAnnotations);
 
-        if (incompatible != null) {
+        if (incompatible.isPresent()) {
```

Comment:
I recommend using `orElseThrow`

Reply:
It uses `isPresent()` so `orElseThrow` won't work

Reply:
you are correct